lib/auth.js

```diff
@@ -417,22 +256,24 @@ api.createIdentity = function(actor, identity, callback) {
 };
 
 /**
- * Gets the Identity ID(s) that match the given email address.
+ * Gets an IdentityID from the URL parameters in the given request. In the
+ * common case, the authenticated user for the request can be used to obtain
+ * the ID without an extra database hit.
  *
- * @param email the email address.
- * @param callback(err, identityIds) called once the operation completes.
+ * @param req the request.
+ * @param callback(err, identityId) called once the operation completes.
  */
-api.resolveEmail = function(email, callback) {
-  database.collections.identity.find(
-    {'identity.email': email},
-    {'identity.id': true}).toArray(function(err, records) {
-    if(records) {
-      records.forEach(function(record, i) {
-        records[i] = record.identity.id;
-      });
-    }
-    callback(err, records);
-  });
+api.getIdentityIdFromUrl = function(req, callback) {
+  // TODO: determine if this function's usage anywhere leaks DID information
+  // through its slug
+
+  // if slug from URL matches authenticated user (common case), then
+  // use its ID
+  var slug = req.params.identity;
+  if(req.isAuthenticated() && req.user && req.user.identity.sysSlug === slug) {
```

Comment:
`req.isAuthenticated()` returns true when `req.user` is truthy -- so we don't need the additional `&& req.user` check.